pkg/mirroring/s3.go

```diff
@@ -0,0 +1,107 @@
+package mirroring
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/warptools/warpforge/wfapi"
+
+	"github.com/serum-errors/go-serum"
+)
+
+type S3Pusher struct {
+	client       *s3.Client
+	cfg          wfapi.S3PushConfig
+	existingKeys map[string]bool
+}
+
+func wareIdToKey(wareId wfapi.WareID) string {
+	return filepath.Join(wareId.Hash[0:3], wareId.Hash[3:6], wareId.Hash)
+}
+
+func newS3Pusher(cfg wfapi.S3PushConfig) (S3Pusher, error) {
+	config, err := config.LoadDefaultConfig(context.TODO(),
```

Comment:
I find that it is usually easier to add `context.Context` to the function APIs sooner rather than later and default to passing it along unless I'm fairly confident that a function is simple enough to not need network, logging, or async. I generally only use `context.TODO` to migrate an old codebase that isn't context aware (which is getting rarer these days).

Reply:
Agree, missed this. Resolved now :)